pkg/epp/scheduling/plugins/plugins.go

```diff
@@ -49,10 +49,11 @@ type Filter interface {
 	Filter(ctx *types.SchedulingContext, pods []types.Pod) []types.Pod
 }
 
-// Scorer defines the interface for scoring pods based on context.
+// Scorer defines the interface for scoring a list of pods based on context.
+// Scorers must score pods with a value within the range of [0,1] where 1 is the highest score.
 type Scorer interface {
 	Plugin
-	Score(ctx *types.SchedulingContext, pod types.Pod) float64
+	Score(ctx *types.SchedulingContext, pods []types.Pod) map[types.Pod]float64
```

Comment:
Agree this new interface is more flexible, however at the cost of more boilerplate code for every new scorer (I guess most scorers will only need to score on per pod basis). If you need access to all pods, the SchedulingCtx object contains that info. So I would at least argue using the ScheudlingCtx as another option.

Reply:
scorers that are calculating in memory can score each pod separately.
scorers that are accessing anything externally need to have the ability to send all pods at once for scoring for enhanced performance.
having the current function signature allows both and I don't think it adds more than 2-3 code lines.
e.g., if a score want to score per pod basis then instead of:
```go
func Score(ctx, pod) {
    ... score pod logic ...
}
``` 

it will look like:
```
func Score(ctx, pods) {
    for _, pod := range pods {
        ... score pod logic ...
    }
}

Reply:
sounds good to me, no strong opinion here, and going with the flexible one is more future proof